packages/core/TextSearch/TextSearchManager.ts

```diff
@@ -21,148 +21,119 @@ export interface SearchScope {
   tracks?: Array<string>
 }
 
-export default (pluginManager: PluginManager) => {
```

Comment:
Is this more of a style and better pattern change?

Reply:
yep basically it is a different style, it shouldn't have any real functional difference. The basic idea I think is that a "factory function that generates a class type" is sort of rare as far as I know. In this refactor, it passes the argument of pluginManager as the constructor to the class instead of making it a context variable or "closure". I don't really suspect it has any functional different, it just seemed a little more straightforward to me

Reply:
cool! okie dokie. I was just curious. :)

Reply:
@teresam856  The "factory" style is usually only necessary when things have to be gotten from the pluginManager before the class can be produced. The usual (maybe the only?) instance of this in our codebase is for MST models where they have to get the pluggable types from the pluginmanager in order to generate the correct MST model